python/lsst/meas/extensions/gaap/_gaap.py

```diff
@@ -388,21 +415,38 @@ def measure(self, measRecord: lsst.afw.table.SourceRecord, exposure: afwImage.Ex
             convolved = result.psfMatchedExposure
             kernelAcf = self._computeKernelAcf(result.psfMatchingKernel)
 
-            for sigma in self.config.sigmas:
+            # Iterate over apertures
+            for sigma in self.config._sigmas:
                 baseName = self.ConfigClass._getGaapResultName(sF, sigma, self.name)
-                if targetSigma >= sigma:
-                    flagKey = measRecord.schema.join(baseName, "flag_bigpsf")
-                    measRecord.set(flagKey, 1)
-                    continue
+                # Calculate the aperture shape and thepre-factor in
+                # Eq. A16 of Kuijken et al. (2015) to scale the flux.
+                # Include an extra factor of 0.5 to undo the normalization
+                # factor of 2 in `computeFixedMomentsFlux`.
+                if sigma == "PsfFlux":
+                    # PSF photometry case
+                    aperSigma2 = targetSigma**2
+                    aperShape = afwGeom.Quadrupole(aperSigma2, aperSigma2, 0.0)
+                    fluxScaling = 1.0  # Eq. A16 of Kuijken et al. (2015)
+                else:
+                    try:
+                        if sigma == "Optimal":  # Hook for DM-29290
+                            # Optimal elliptical aperture case
+                            # TODO: DM-29290
+                            # Set `normalize` to True to check if aperShape is valid.
+                            continue
+                        else:
+                            aperSigma2 = sigma**2 - targetSigma**2
+                            aperShape = afwGeom.Quadrupole(aperSigma2, aperSigma2, 0.0,
+                                                           normalize=True)
+                            fluxScaling = 0.5*sigma**2/aperSigma2
+                    except (InvalidParameterError, ZeroDivisionError):
+                        # Raised when the aperture is invalid
+                        flagKey = measRecord.schema.join(baseName, "flag_bigpsf")
+                        measRecord.set(flagKey, 1)
+                        continue
 
-                aperSigma2 = sigma**2 - targetSigma**2
-                aperShape = afwGeom.Quadrupole(aperSigma2, aperSigma2, 0.0)
                 fluxResult = measBase.SdssShapeAlgorithm.computeFixedMomentsFlux(convolved.getMaskedImage(),
```

Comment:
It seems like you could call `convolved.getMaskedImage()` once before the loop if it's not changing anywhere inside it.

Reply:
Yes, I now redefined `convolved` to be a `MaskedImage` instead of an `Exposure`.